api/src/wfl/mime_type.clj

```diff
@@ -1,17 +1,29 @@
 (ns wfl.mime-type
   (:require [clojure.string :as str]
-            [ring.util.mime-type :as mime-type]))
+            [ring.util.mime-type :as mime-type]
+            [wfl.util :as util]))
 
 (def ^:private mime-types
   "mime-types for file extensions commonly used in computational biology."
-  (let [bio-mimes {"bam"   "application/octet-stream"
-                   "cram"  "application/octet-stream"
-                   "fasta" "application/octet-stream"
-                   "vcf"   "text/plain"}]
-    (merge mime-type/default-mime-types bio-mimes)))
+  (let [extensions
+        {"bam"     "application/octet-stream"
+         "cram"    "application/octet-stream"
+         "fasta"   "application/octet-stream"
+         "genbank" "application/octet-stream"
+         "ready"   "text/plain"
+         "tsv"     "text/tab-separated-values"
+         "vcf"     "text/plain"}]
+    (merge mime-type/default-mime-types extensions)))
+
+;; visible for testing
+(defn ext-mime-type-no-default
+  "Look up the mime-type of the filename by file extension."
+  [filename]
+  (loop [filename (util/basename filename)]
+    (if-let [ext (util/extension filename)]
+      (or (mime-types ext) (recur (util/remove-extension filename))))))
```

Comment:
just curious about the usage of `loop recur` here, is it for some use cases like `(ext-mime-type-no-default "a.cram.unknown")`?

Reply:
I've see files like "biosample_attributes-merged.genbank.src" in the outputs. Now i have no idea what a `.src` file is, but I know what a `genbank` file is, so if I strip the .src extension off, I'll converge on genbank and return its mime-type by successively trying to find match extensions. Likewise, I have no idea what a .tmp file is, but if it has a sensible file extension before .tmp, it's likely that it's one of them (e.g. cram.tmp is likely a cram).
This might actually be the wrong thing to do.

Reply:
So yes.